core/libraries/batch/JobHandlers/RegistrationsReport.php

```diff
@@ -308,81 +302,11 @@ public function get_csv_data_for($event_id, $offset, $limit, $question_labels, $
-                    }
-                } else {
-                    $value = '';
-                }
-                $reg_csv_array[ EEH_Export::get_column_name_for_field($field_obj) ] = $value;
-            }
-            $is_primary_reg = $reg_row['Registration.REG_count'] == '1' ? true : false;
-            /*@var $reg_row EE_Registration */
-            foreach ($reg_fields_to_include as $field_name) {
-                $field = $reg_model->field_settings_for($field_name);
-                if ($field_name == 'REG_final_price') {
-                    $value = EEH_Export::prepare_value_from_db_for_display(
-                        $reg_model,
-                        $field_name,
-                        $reg_row['Registration.REG_final_price'],
-                        'localized_float'
-                    );
-                } elseif ($field_name == 'REG_count') {
-                    $value = sprintf(
-                        /* translators: 1: number of registration in group (REG_count), 2: registration group size (REG_group_size) */
-                        esc_html__('%1$s of %2$s', 'event_espresso'),
-                        EEH_Export::prepare_value_from_db_for_display(
-                            $reg_model,
-                            'REG_count',
-                            $reg_row['Registration.REG_count']
-                        ),
-                        EEH_Export::prepare_value_from_db_for_display(
-                            $reg_model,
-                            'REG_group_size',
-                            $reg_row['Registration.REG_group_size']
-                        )
-                    );
-                } elseif ($field_name == 'REG_date') {
-                    $value = EEH_Export::prepare_value_from_db_for_display(
-                        $reg_model,
-                        $field_name,
-                        $reg_row['Registration.REG_date'],
-                        'no_html'
-                    );
-                } else {
-                    $value = EEH_Export::prepare_value_from_db_for_display(
-                        $reg_model,
-                        $field_name,
-                        $reg_row[ $field->get_qualified_column() ]
-                    );
-                }
-                $reg_csv_array[ EEH_Export::get_column_name_for_field($field) ] = $value;
-                if ($field_name == 'REG_final_price') {
-                    // add a column named Currency after the final price
-                    $reg_csv_array[ (string) esc_html__("Currency", "event_espresso") ] = EE_Config::instance()->currency->code;
-                }
-            }
+            $reg_csv_array = (new Attendee($att_fields_to_include, $reg_row))
+                ->addAttendeeColumns($reg_csv_array);
+            // add registration columns
+            $reg_csv_array = (new Registration($reg_fields_to_include, $reg_row))
+                ->addRegistrationColumns($reg_csv_array);
```

Comment:
let's split these up and verify the instance so that we don't violate the "golden rule" of programming

> "If it can be `null`, it will be `null`

As well, I'm thinking we should maybe clarify the names of these new classes a bit to differentiate them from regular entities, ie: `AttendeeCSV` and `RegistrationCSV`

```php
            $attendee_csv = new AttendeeCSV($att_fields_to_include, $reg_row);
            $reg_csv_array = $attendee_csv instanceof AttendeeCSV 
		? $attendee_csv->addAttendeeColumns($reg_csv_array)
		: $reg_csv_array;
            // add registration columns
	    $registration_csv = new RegistrationCSV($reg_fields_to_include, $reg_row);
            $reg_csv_array = $registration_csv instanceof RegistrationCSV
		? $registration_csv->addRegistrationColumns($reg_csv_array)
		: $reg_csv_array;
```